packages/main/src/resolvers/conversation.ts

```diff
@@ -118,9 +118,10 @@ export const ConversationMutations: ConversationMutationsResolvers = {
 
   async flagPresentable(_parent, { conversationId, id, isFlagged }) {
     const thread = C.mustGetConversation(conversationId)
-    for (const message of thread.messages) {
-      if (String(message.id) === id) {
-        updateAction([message], (accountId, box, uids) => {
+    const presentables = C.getPresentableElements(thread)
+    for (const presentable of presentables) {
+      if (presentable.id === id) {
+        updateAction(presentable.revisions, (accountId, box, uids) => {
```

Comment:
This is better! But when adding a flag I would like to flag the last revision instead of all revisions. When removing flags we should filter to revisions that are flagged, and just remove flags from those.